src/main/java/com/commercetools/sync/categories/helpers/CategoryAssetActionFactory.java

```diff
@@ -3,29 +3,36 @@
 import com.commercetools.sync.categories.CategorySyncOptions;
 import com.commercetools.sync.commons.helpers.AssetActionFactory;
 import io.sphere.sdk.categories.Category;
+import io.sphere.sdk.categories.CategoryDraft;
 import io.sphere.sdk.categories.commands.updateactions.AddAsset;
 import io.sphere.sdk.categories.commands.updateactions.ChangeAssetOrder;
 import io.sphere.sdk.categories.commands.updateactions.RemoveAsset;
 import io.sphere.sdk.commands.UpdateAction;
 import io.sphere.sdk.models.Asset;
 import io.sphere.sdk.models.AssetDraft;
+import io.sphere.sdk.models.Resource;
 
 import javax.annotation.Nonnull;
 import java.util.List;
 
 import static com.commercetools.sync.categories.utils.CategoryAssetUpdateActionUtils.buildActions;
 
-public final class CategoryAssetActionFactory extends AssetActionFactory<Category> {
+public final class CategoryAssetActionFactory<T,D extends CategoryDraft>
+    extends AssetActionFactory<Category> {
```

Comment:
I would rather expect here sth like `extends AssetActionFactory<Category, CategoryDraft>` and in the parent class you define it as `public abstract class AssetActionFactory<T, D> {`.

Reply:
3ca94f89